third_party/nvfuser/test/test_gpu2.cpp

```diff
@@ -7133,6 +7133,7 @@ TEST_F(NVFuserTest, FusionPredicateElimination6_CUDA) {
   // for the on on split root domains. As the second root axis of tv2
   // is split, its index exceeds its extent (i.e., 3 in this case)
   // without its predicate.
+  // The following check fails with eager constant ops
   TORCH_CHECK(PredicatedChecker::isPredicated(tv2, gpulw));
```

Comment:
This test fails. This is the IR dump before this PR:
```

Inputs:
  T0_g[ iS0{2}, iS18{( ceilDiv(3, 5) )}, iS19{5} ], float
Outputs:
  T4_g[ iS10{( ceilDiv(3, 5) )}, iS8{2}, iS11{5} ] produce_pos( 1 ), float

%kernel_math {
T1_l[ iS2{2}, iS16{( ceilDiv(3, 5) )}, iS17{5} ]
   = T0_g[ iS0{2}, iS18{( ceilDiv(3, 5) )}, iS19{5} ]
   + double(1);
T2_l[ iS4{2}, iS14{( ceilDiv(3, 5) )}, iS15{5} ]
   = T1_l[ iS2{2}, iS16{( ceilDiv(3, 5) )}, iS17{5} ]
   + double(1);
T3_l[ iS12{( ceilDiv(3, 5) )}, iS13{5}, iS6{2} ] ca_pos( 1 )
   = T2_l[ iS4{2}, iS14{( ceilDiv(3, 5) )}, iS15{5} ]
   + double(1);
T4_g[ iS10{( ceilDiv(3, 5) )}, iS8{2}, iS11{5} ] produce_pos( 1 )
   = T3_l[ iS12{( ceilDiv(3, 5) )}, iS13{5}, iS6{2} ] ca_pos( 1 )
   + double(1);
}
```
and after:
```
Inputs:
  T0_g[ iS0{2}, iS18{1}, iS19{5} ], float
Outputs:
  T4_g[ iS10{1}, iS8{2}, iS11{5} ] produce_pos( 1 ), float

%kernel_math {
T1_l[ iS2{2}, iS16{1}, iS17{5} ]
   = T0_g[ iS0{2}, iS18{1}, iS19{5} ]
   + double(1);
T2_l[ iS4{2}, iS14{1}, iS15{5} ]
   = T1_l[ iS2{2}, iS16{1}, iS17{5} ]
   + double(1);
T3_l[ iS12{1}, iS13{5}, iS6{2} ] ca_pos( 1 ) 
   = T2_l[ iS4{2}, iS14{1}, iS15{5} ]
   + double(1);
T4_g[ iS10{1}, iS8{2}, iS11{5} ] produce_pos( 1 )
   = T3_l[ iS12{1}, iS13{5}, iS6{2} ] ca_pos( 1 )
   + double(1);
}
```
I believe this is just a problem with how the `PredicatedChecker` works. Currently it just tests whether a value is constant to determine if it is predicated. Before this PR, the extent after split is a symbolic `ceilDiv(3, 5)` which is non-const, but this PR makes it const.

Unfortunately, this is a real issue as `compute-sanitizer --tool initcheck` currently shows uninitialized reads with this PR. Before merging, I need to revisit the logic for inserting predicates.